parsers/julia.c

```diff
@@ -1305,8 +1398,10 @@ static void parseExpr (lexerState *lexer, bool delim, int kind, vString *scope)
                 parseType(lexer, scope, kind);
                 break;
             case TOKEN_IMPORT:
-                parseImport(lexer, scope, kind);
+                parseImport(lexer, scope, TOKEN_IMPORT);
```

Comment:
Just a question: can we expect "import" and "using"  always appear at the top-level in an input source code?

I don't know Julia but I made a pseudo example:
```
function foo
{
    import Library1
}
```
This is an example that "import" doesn't appear at the top-level.

Reply:
> can we expect "import" and "using" always appear at the top-level in an input source code?

There is no clear answer, but I think it's pretty safe to assume this.

The common use of `import` or `using` "inside something" is inside a module:

``` julia
module MyModule
using Base64
# This will work
println(Base64.base64encode("hi"))
end

# This will throw an error
println(Base64.base64encode("hi"))
```

In the terminology of Julia, "inside a module" is still considered "top level", after all it doesn't have a file-level scope. Current parser works fine on this.

Put it in a function will cause error:

``` julia
function f()
    using Base64
end
# => ERROR: LoadError: syntax: "using" expression not at top level
```

This suggests that julia wants to ensure `using` is at top level. But actually it isn't very strict on this.

``` julia
let a = 1
    using Base64
end
```

`let`, like `function` creates a scope. But the above code runs well.

Even inside a function, we could still use some trick:

``` julia
function f()
    @eval using Base64
end
```

This runs well.

`import` is the same as above.

Unfortunately the language reference is also not very clear on this, and I don't know the implementation of `using` or `import`. But based on common practice, I think it's pretty safe to assume `using` and `import` are at top level.

Reply:
I see.